reconcile/utils/slack_api.py

```diff
@@ -322,29 +356,30 @@ def get_user_id_by_name(self, user_name: str, mail_address: str) -> str:
 
         return result["user"]["id"]
 
-    def get_channels_by_names(self, channels_names):
+    def get_channels_by_names(self, channels_names: Iterable[str]) -> dict[str, str]:
         return {
             k: v["name"]
             for k, v in self._get("channels").items()
             if v["name"] in channels_names
         }
 
-    def get_channels_by_ids(self, channels_ids):
+    def get_channels_by_ids(self, channels_ids: Iterable[str]) -> dict[str, str]:
         return {
             k: v["name"] for k, v in self._get("channels").items() if k in channels_ids
         }
 
-    def get_users_by_names(self, user_names):
+    def get_users_by_names(self, user_names: Iterable[str]) -> dict[str, str]:
         return {
             k: v["name"]
             for k, v in self._get("users").items()
             if v["name"] in user_names
         }
 
-    def get_users_by_ids(self, users_ids):
+    def get_users_by_ids(self, users_ids: Iterable[str]) -> dict[str, str]:
         return {k: v["name"] for k, v in self._get("users").items() if k in users_ids}
 
-    def _get(self, resource: str) -> Dict[str, Any]:
+    @lru_cache(maxsize=10)
```

Comment:
There is already some caching going on in this function using a dict `_results`
Which benefit does the LRU Cache provide?

Reply:
I've totally overseen this 🤦